[quantization] JIT serialization for Conv2d

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24117 [quantization] JIT serialization for Conv2d**
* #24116 [quantization] state_dict serialization for Conv2d + some bugfixes
* #24115 [quantization] Re-work Conv2d

Differential Revision: [D16765475](https://our.internmc.facebook.com/intern/diff/D16765475)

Commit:
[quantization] JIT serialization for Conv2d

Commit:
Update on "[quantization] JIT serialization for Conv2d"

raghuramank100 commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -116,6 +116,25 @@ def _save_to_state_dict(self, destination, prefix, keep_vars):
+            self.groups,
+            self.padding_mode,
+            self.weight(),
+            self.bias,
```
This can be none. Is special handling requires to handle that?

jamesr66a replied:
No, I don't know why we had special handling before, `None`s are directly representable in both the Python and TorchScript pickle formats

Commit:
Update on "[quantization] JIT serialization for Conv2d"

Commit:
Update on "[quantization] JIT serialization for Conv2d"


Differential Revision: [D16765475](https://our.internmc.facebook.com/intern/diff/D16765475)

Commit:
Update on "[quantization] JIT serialization for Conv2d"


Differential Revision: [D16765475](https://our.internmc.facebook.com/intern/diff/D16765475)
